types/govuk-frontend/dist/govuk/govuk-frontend-component.d.ts

```diff
@@ -2,8 +2,38 @@
  * Base Component class
  *
  * Centralises the behaviours shared by our components
- *
- * @internal
  */
-export class GOVUKFrontendComponent {
+export abstract class GOVUKFrontendComponent<RootElementType extends Element = HTMLElement> {
+    static elementType: typeof Element;
+
+    /**
+     * Validates whether components are supported
+     *
+     * @throws {SupportError} when the components are not supported
+     */
+    static checkSupport(): void;
+
+    /**
+     * Constructs a new component, validating that GOV.UK Frontend is supported
+     *
+     * @internal
+     * @param {Element | null} [$root] - HTML element to use for component
+     */
+    constructor($root?: Element | null);
+
+    /**
+     * Returns the root element of the component
+     *
+     * @returns {RootElementType} - the root element of component
+     */
+    protected get $root(): RootElementType;
```

Comment:
@romaricpascal There was also `protected _$root` but I've taken it out

I think you meant to add `@private` not `@internal` in your JSDoc?

Otherwise let me know and I'll add it back here

Reply:
Yeah, I think it can be left out from this package. If people are using `@types/govuk-frontend` TypeScript will know that `this.$root` is available in the child classes so don't need `this._$root`. 

`this._$root` was made available so people could implement their own getter rather than sprinkle `@ts-expect-error` everytime they'd want to access `this.$root` when TypeScript doesn't have the types for `govuk-frontend`.

I'll make a note that we can move it to `@private`.

Reply:
Great minds, exactly why I left it out